packages/plugin-link/src/Link.ts

```diff
@@ -36,57 +41,51 @@ export class Link<T extends JWPluginConfig = JWPluginConfig> extends JWPlugin<T>
             handler: this.unlink,
         },
     };
-    readonly loadables: Loadables<Parser & Keymap> = {
+    readonly loadables: Loadables<Parser & Keymap & Layout & Owl> = {
         parsers: [LinkXmlDomParser],
         shortcuts: [
             {
                 pattern: 'CTRL+K',
                 selector: [(node: VNode): boolean => !Link.isLink(node)],
                 commandId: 'link',
-                // TODO: use dialog to get params
-                commandArgs: {
-                    url: 'https://en.wikipedia.org/wiki/Jabberwocky',
-                    label: 'Jabberwockipedia',
-                },
             },
             {
                 pattern: 'CTRL+K',
                 selector: [Link.isLink],
                 commandId: 'unlink',
             },
         ],
+        components: [
+            {
+                id: 'link',
+                async render(): Promise<OwlNode[]> {
+                    return [new OwlNode(LinkComponent, {})];
+                },
+            },
+        ],
+        owlTemplates: [linkForm],
     };
 
     //--------------------------------------------------------------------------
     // Public
     //--------------------------------------------------------------------------
 
-    link(params: LinkParams): void {
-        const range = params.context.range;
-        const selectedInlines = range.selectedNodes(InlineNode);
+    async link(params: LinkParams): Promise<void> {
+        // If the url is undefined, ask the user to provide one.
+        if (!params.url) {
+            const layoutPlugin = this.editor.plugins.get(Layout);
```

Comment:
Same as `KeyMap`. There is no difference between the `Layout` *plugin* and *the* layout. The `Layout` plugin *is* the layout. So I would just call this variable `Layout`.

Reply:
It is an instance though right? So `layout` is more appropriate.